### PR TITLE
Add 2fa tests

### PR DESCRIPTION
**Pull request by otaviojacobi**
Change-type: patch

Resolves: # 
HQ: <url> 
See: <url> 
Depends-on: <url> 
Change-type: major|minor|patch 

---
##### Contributor checklist

- [ ] Includes tests
- [ ] Includes typings
- [ ] Includes updated documentation
- [ ] Includes updated build output

### CODE REVIEW

File: .github/actions/test/action.yml [hunk 1]
```diff
@@ -52,6 +52,9 @@ runs:
         export TEST_MEMBER_USERNAME=${{ fromJSON(inputs.secrets)[ format('TEST_MEMBER_USERNAME{0}', fromJSON('{"windows-2019":"","ubuntu-20.04":"_1","macos-12":"_2"}')[env.os_value || 'windows-2019' ]) ] }}
         export TEST_MEMBER_PASSWORD=${{ fromJSON(inputs.secrets).TEST_MEMBER_PASSWORD }}
         export TEST_ONLY_ON_ENVIRONMENT=${{ fromJSON('{"windows-2019":"node","ubuntu-20.04":"node","macos-12":"browser"}')[env.os_value || 'windows-2019' ]  }}
+        export TEST_2FA_EMAIL=${{ fromJSON(inputs.secrets)[ format('TEST_2FA_EMAIL{0}', fromJSON('{"windows-2019":"","ubuntu-20.04":"_1","macos-12":"_2"}')[env.os_value || 'windows-2019' ]) ] }}
```
Comment (otaviojacobi) [inline .github/actions/test/action.yml:55]: @thgreasi do you have permissions to set these environment variables on this repo?

Comment (thgreasi) [inline .github/actions/test/action.yml:55]: Given that we define a single secret for all runners, which is autogenerated by the API, I think accepting different emails per runner doesn't make sense.
```suggestion
        export TEST_2FA_EMAIL=${{ fromJSON(inputs.secrets).TEST_2FA_EMAIL }}
```

Comment (thgreasi) [reply .github/actions/test/action.yml:55]: Created a user and added the secrets :+1:

File: tests/integration/auth.spec.js [hunk 2]
```diff
@@ -324,4 +326,56 @@ describe('SDK authentication', function () {
...
+			);
+
+			given2FAUserIt(
+				'balena.auth.twoFactor.isPassed() should be true when 2FA is passed',
+				async () => {
+					const code = authenticator.generate(credentials.twoFactor.secret);
+					await balena.auth.twoFactor.challenge(code);
+					expect(await balena.auth.twoFactor.isPassed()).to.be.true;
+				},
+			);
```
Comment (thgreasi) [inline tests/integration/auth.spec.js:378]: NIT for consistency
```suggestion
			describe('balena.auth.twoFactor.isEnabled()', function () {
				given2FAUserIt(
					'should be true',
					async () => {
						expect(await balena.auth.twoFactor.isEnabled()).to.be.true;
					},
				);
			});

			describe('balena.auth.twoFactor.isPassed()', function () {
				given2FAUserIt(
					'should false when 2FA is not passed',
					async () => {
						expect(await balena.auth.twoFactor.isPassed()).to.be.false;
					},
				);

				given2FAUserIt(
					'should be true when 2FA is passed',
					async () => {
						const code = authenticator.generate(credentials.twoFactor.secret);
						await balena.auth.twoFactor.challenge(code);
						expect(await balena.auth.twoFactor.isPassed()).to.be.true;
					},
				);
			});
```

**Review by thgreasi — COMMENTED** [review]
other than those, feel free to merge it

**Pull request merged**